engine/apps/alerts/models/alert_receive_channel_connection.py

```diff
@@ -18,3 +18,19 @@ class AlertReceiveChannelConnection(models.Model):
     class Meta:
         ordering = ["source_alert_receive_channel", "connected_alert_receive_channel"]
         unique_together = ("source_alert_receive_channel", "connected_alert_receive_channel")
+
+
+class AlertGroupExternalID(models.Model):
+    """
+    This model represents an external ID for an alert group. This is used to keep track of the alert group in
+    the external system (e.g. ServiceNow).
+    """
+
+    source_alert_receive_channel = models.ForeignKey(
+        "AlertReceiveChannel", on_delete=models.CASCADE, related_name="external_ids"
+    )
+    alert_group = models.ForeignKey("AlertGroup", on_delete=models.CASCADE, related_name="external_ids")
+    value = models.CharField(max_length=512)
+
+    class Meta:
+        unique_together = ("source_alert_receive_channel", "alert_group")
```

Comment:
Should we have an index for the value? (I guess we will be querying for the alert group using the external ID and the source channel integration type?)

Reply:
done in https://github.com/grafana/oncall/pull/4076/commits/5433ab60744048aef71c9625f7c10c37479fd362